dm/ui/src/models/source.ts

```diff
@@ -0,0 +1,22 @@
+export interface Source {
```

Comment:
maybe we can use openapi spec to generate those models 
https://swagger.io/tools/swagger-codegen/ the spec is [here](https://github.com/pingcap/ticdc/blob/master/dm/openapi/spec/dm.yaml)

Reply:
This is how tidb-dashboard does it if i recall correctely and I honestly don't think this is necessary here. It's slow, and requires java environment . And the generated code is a whole api client, making ui code heavy coupled with it and very difficult to reuse ui component else where.

Reply:
also it generate code based on a openapi format json, but the openapi spec itself lacks enough infomation to generate solid, type-safe typescript code

Reply:
sgtm